lib/formats/bs_format_ripejson.c

```diff
@@ -0,0 +1,669 @@
+
+void process_common_fields(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  // populate collector name
+  memcpy(record -> collector_name , STATE->json_fields.host.ptr, STATE->json_fields.host.len);
+  record -> collector_name[STATE->json_fields.host.len] = '\0';
+
+  // populate peer asn
+  RDATA->elem->peer_asn = (uint32_t) strtol((char *)STATE->json_fields.peer_asn.ptr, NULL, 10);
+
+  // populate peer ip
+  bgpstream_addr_storage_t addr;
+  memcpy(STATE->field_buffer, STATE->json_fields.peer.ptr, STATE->json_fields.peer.len);
+  STATE->field_buffer[STATE->json_fields.peer.len] = '\0';
+  bgpstream_str2addr((char *)STATE->field_buffer, &addr);
+  bgpstream_addr_copy((bgpstream_ip_addr_t *)&RDATA->elem->peer_ip,
+                      (bgpstream_ip_addr_t *)&addr);
+  // populate time-stamp
+  double time_double = strtod((char *)STATE->json_fields.timestamp.ptr, NULL);
+  record->time_sec = (int) time_double;
+  record->time_usec = (int)((time_double - (int) time_double) * 1000000);
+
+}
+
+int process_update_message(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  // convert body to bytes
+  size_t dec_len =  (size_t) hexstr_to_bgpmsg(STATE->json_bytes_buffer,
+                                      4096,
+                                      (char*) STATE->json_fields.body.ptr,
+                                      STATE->json_fields.body.len,
+                                      PARSEBGP_BGP_TYPE_UPDATE);
+  size_t err;
+
+  RDATA->msg->type = PARSEBGP_MSG_TYPE_BGP;
+  if ((err = parsebgp_decode(STATE->opts, RDATA->msg->type, RDATA->msg,
+                             STATE->json_bytes_buffer, &dec_len)) != PARSEBGP_OK) {
+    fprintf(stderr, "ERROR: Failed to parse message (%zu:%s)\n", err, parsebgp_strerror(err));
+    parsebgp_clear_msg(RDATA->msg);
+    return -1;
+  }
+
+  process_common_fields(format, record);
+
+  return 0;
+}
+
+int process_status_message(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  // process direction
+  if( bcmp(&"down", STATE->json_fields.state.ptr, STATE->json_fields.state.len) == 0){
+    // equals
+    RDATA->status_msg_state = 0;
+  } else if( bcmp(&"connected", STATE->json_fields.state.ptr, STATE->json_fields.state.len) == 0){
+    RDATA->status_msg_state = 1;
+  } else {
+    // unknown
+    RDATA->status_msg_state = -1;
+  }
+
+  process_common_fields(format, record);
+
+  return 0;
+}
+
+int process_open_message(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  int json_str_len = STATE->json_fields.body.len;
+  uint8_t* json_str_ptr = STATE->json_fields.body.ptr;
+  int missing_type = 0;
+  int err;
+  uint8_t* ptr = STATE->json_bytes_buffer;
+  int loc = 2;
+  int msg_len = 0;
+
+
+  ptr[loc] = PARSEBGP_BGP_TYPE_OPEN;
+  loc += 1;
+
+  /* add missing open message headers */
+
+  // version
+  ptr[loc] = 4;
+  loc += 1;
+
+  // my autonomous system
+  uint32_t asn4 = strtol((char*)STATE->json_fields.asn.ptr, NULL, 10);
```

Comment:
`strtoul` i think.
also, the json string pointers are not nul-terminated after the value. i guess you're relying on strtoul stopping parsing when it gets to a json character (e.g., a comma). this might work but it worries me a little.